interop/grpclb_fallback/client.go

```diff
@@ -0,0 +1,217 @@
+	}
+	req := &testpb.SimpleRequest{
+		FillGrpclbRouteType: true,
+	}
+	reply, err := client.UnaryCall(ctx, req, copts...)
+	if err != nil {
+		grpclog.Infof("doRPCAndGetPath error:%v", err)
+		return testpb.GrpclbRouteType_GRPCLB_ROUTE_TYPE_UNKNOWN
+	}
+	g := reply.GetGrpclbRouteType()
+	grpclog.Infof("doRPCAndGetPath got grpclb route type: %v", g)
+	if g != testpb.GrpclbRouteType_GRPCLB_ROUTE_TYPE_FALLBACK && g != testpb.GrpclbRouteType_GRPCLB_ROUTE_TYPE_BACKEND {
+		grpclog.Fatalf("Expected grpclb route type to be either backend or fallback", g)
+	}
+	return g
+}
+
+func dialTCPUserTimeout(ctx context.Context, addr string) (net.Conn, error) {
+	control := func(network, address string, c syscall.RawConn) error {
+		var syscallErr error
+		controlErr := c.Control(func(fd uintptr) {
+			syscallErr = syscall.SetsockoptInt(int(fd), syscall.IPPROTO_TCP, unix.TCP_USER_TIMEOUT, 20000)
+		})
+		if syscallErr != nil {
+			grpclog.Fatalf("syscall error setting sockopt TCP_USER_TIMEOUT: %v", syscallErr)
+		}
+		if controlErr != nil {
+			grpclog.Fatalf("control error setting sockopt TCP_USER_TIMEOUT: %v", syscallErr)
+		}
+		return nil
+	}
+	d := &net.Dialer{
+		Control: control,
+	}
+	return d.DialContext(ctx, "tcp", addr)
+}
+
+func createFallbackTestConn() *grpc.ClientConn {
+	opts := []grpc.DialOption{
+		grpc.WithContextDialer(dialTCPUserTimeout),
+		grpc.WithBlock(),
+	}
+	switch *customCredentialsType {
+	case "tls":
+		creds := credentials.NewClientTLSFromCert(nil, "")
+		opts = append(opts, grpc.WithTransportCredentials(creds))
+	case "alts":
+		creds := alts.NewClientCreds(alts.DefaultClientOptions())
+		opts = append(opts, grpc.WithTransportCredentials(creds))
+	case "google_default_credentials":
+		opts = append(opts, grpc.WithCredentialsBundle(google.NewDefaultCredentials()))
+	case "compute_engine_channel_creds":
+		opts = append(opts, grpc.WithCredentialsBundle(google.NewComputeEngineCredentials()))
+	default:
+		grpclog.Fatalf("Invalid --custom_credentials_type:%v", *customCredentialsType)
+	}
+	conn, err := grpc.Dial(*serverURI, opts...)
+	if err != nil {
+		log.Fatalf("Fail to dial: %v", err)
+	}
+	return conn
+}
+
+func runCmd(command string) {
+	grpclog.Infof("Running cmd:|%v|", command)
+	if err := exec.Command("bash", "-c", command).Run(); err != nil {
+		grpclog.Fatalf("error running cmd:|%v| : %v", command, err)
+	}
+}
+
+func runFallbackBeforeStartupTest(breakLBAndBackendConnsCmd string, perRPCDeadlineSeconds int) {
+	conn := createFallbackTestConn()
+	defer conn.Close()
+	client := testpb.NewTestServiceClient(conn)
+	runCmd(breakLBAndBackendConnsCmd)
+	for i := 0; i < 30; i++ {
+		if g := doRPCAndGetPath(client, perRPCDeadlineSeconds, failFast); g != testpb.GrpclbRouteType_GRPCLB_ROUTE_TYPE_FALLBACK {
+			grpclog.Fatalf("Expected RPC to take grpclb route type FALLBACK. Got: %v", g)
+		}
+		time.Sleep(time.Second)
+	}
+}
+
+func doFastFallbackBeforeStartup() {
+	runFallbackBeforeStartupTest(*unrouteLBAndBackendAddrsCmd, 9)
```

Comment:
Do you want to make `9` and `20` configurable?

Reply:
These are defined in the spec - I don't think they need to be configurable for now